packages/snap-controller/src/Autocomplete/AutocompleteController.test.ts

```diff
@@ -221,21 +275,26 @@ describe('Autocomplete Controller', () => {
 			logger: new Logger(),
 			tracker: new Tracker(globals),
 		});
-		await controller.init();
-		const setFocusedfn = jest.spyOn(controller, 'setFocused');
+		let setFocusedfn: any;
 
-		const inputEl: HTMLInputElement | null = document.querySelector(controller.config.selector);
-		const query = 'bumpers';
-		inputEl!.value = query;
-		inputEl!.focus();
+		await waitFor(() => {
+			controller.init();
+			setFocusedfn = jest.spyOn(controller, 'setFocused');
 
-		expect(controller.urlManager.state.query).toBe(undefined);
-		expect(setFocusedfn).toHaveBeenCalledTimes(0);
+			const inputEl: HTMLInputElement | null = document.querySelector(controller.config.selector);
+			const query = 'bumpers';
+			inputEl!.value = query;
+			inputEl!.focus();
 
-		await controller.bind();
-		expect(setFocusedfn).toHaveBeenCalledTimes(1);
+			expect(controller.urlManager.state.query).toBe(undefined);
+			expect(setFocusedfn).toHaveBeenCalledTimes(0);
+		});
 
-		setFocusedfn.mockClear();
+		await waitFor(() => {
+			controller.bind();
```

Comment:
This could potentially call `bind()` over and over. Should pull non-assertion out of `waitFor`.